### PR TITLE
Implement span exporting limits

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java [hunk 1]
```diff
@@ -143,7 +144,13 @@ SpanExporter buildExporter(ConnectionUtil connectionUtil) {
             // we'll do our best to hang on to the spans with the wrapping BufferingExporter.
             ZipkinSpanExporter.baseLogger.setLevel(Level.SEVERE);
         }
-        return new BufferingExporter(connectionUtil, ZipkinSpanExporter.builder().setEndpoint(endpoint).build());
+        ZipkinSpanExporter zipkinSpanExporter = ZipkinSpanExporter.builder().setEndpoint(endpoint).build();
+        SpanExporter throttlingExporter = ThrottlingExporter.newBuilder(zipkinSpanExporter)
+                .categorizeByAttribute(SplunkRum.COMPONENT_KEY)
+                .maxSpansInWindow(100)
+                .windowSize(Duration.ofSeconds(30))
```
Comment (mateuszrzeszutek) [inline splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java:151]: @jkwatson should I add these to `Config`? I feel kind of bad hard-coding them here...

Comment (jkwatson) [reply splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java:151]: @johnbley is this something iOS has made configurable?

**Comment by jkwatson** [comment]
What do you think about adding a button to the demo app which will exercise this functionality?

File: splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java (hunk 1 above)
Comment (johnbley) [reply splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java:151]: Not currently; I don't expect it would be necessary.

**Comment by mateuszrzeszutek** [comment]
> What do you think about adding a button to the demo app which will exercise this functionality?

Good point - done. I added a "spam button" to the second fragment

File: splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java (hunk 1 above)
Comment (jkwatson) [reply splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java:151]: let's leave them out for now. If someone needs it, we can always add it later.

**Review by jkwatson — APPROVED** [review]
Thanks! I think this is slightly more sophisticated than what's in iOS right now, but I think that's totally fine.

**Pull request merged**